xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/StringClass.java

```diff
@@ -123,4 +124,17 @@ public void displayEdit(StringBuffer buffer, String name, String prefix, BaseCol
         buffer.append(input.toString());
     }
 
+    @Override
+    public void displayView(StringBuffer buffer, String name, String prefix, BaseCollection object,
+        XWikiContext context)
+    {
+        BaseProperty prop = (BaseProperty) object.safeget(name);
+        String result;
+        if (prop != null) {
+            result = prop.toText();
+        } else {
+            result = "";
+        }
```

Comment:
Small note that there seems to be code duplication. I see the same logic at https://github.com/xwiki/xwiki-platform/pull/1315/files#diff-912dd4275a5ac5f2116abef20711b489R384

Reply:
Safe access to the property's text factored here https://github.com/xwiki/xwiki-platform/pull/1315/files#diff-8574f36522a5e8eab9ad2acf469c1ef0R140

Reply:
Reverted back to avoid the introduction of a undesired public API